validation-test/Sema/issue60958.swift

```diff
@@ -0,0 +1,36 @@
+// RUN: %target-swift-frontend -emit-silgen %s | %FileCheck %s
+
+// https://github.com/apple/swift/issues/60958
+// Make sure that `makeIterator` witness is picked over the non-witness.
+
+struct S: Sequence {
```

Comment:
@xedin Could be worth adding a test case that involves a generic argument?
```swift
struct G<T>: Sequence {
  private var _storage: [T] = []
  func makeIterator() -> Array<T>.Iterator {
    _storage.makeIterator()
  }
  
  typealias Element = T
  class Iterator: IteratorProtocol {
    func next() -> T? { fatalError() }
    typealias Element = T
  }

  func makeIterator() -> Iterator {
    Iterator()
  }
}

extension G {
  func f() {
    for elt in self { // OK
      print(elt)
    }
  }
}
```

Reply:
Other test-case is generic actually :)

Reply:
Ah true, miss that 😅